databases/week3/mealSharing.sql

```diff
@@ -0,0 +1,149 @@
+
+--Get a meal with any id, fx 1
+SELECT * FROM meal
+WHERE meal.id = 3;
+
+--Update a meal with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE meal 
+SET title = 'Paneer Buttur masala', description = 'Cottage cheese in rich creamy sauce with onions and tomato puree'
+WHERE id = 3;
+
+--Delete a meal with any id, fx 1
+DELETE FROM meal
+WHERE id = 4;
+
+--Reservation
+--Add a new reservation
+INSERT INTO reservation (number_of_guests, meal_id, created_date, contact_phonenumber, contact_name, contact_email)
+VALUES (9, 2, '2022-06-30', 91919191, 'Manisha Patel', 'manisha@gmail.com'),
+(5, 1, '2022-07-30', 91676767, 'Chirag', 'chirag@gmail.com'),
+(6, 3, '2022-05-28', 91546545, 'Sachi Patel', 'sachi@gmail.com'),
+(8, 2, '2022-07-13', 91898989, 'Jiaan', 'jiaan@gmail.com');
+
+--Get a reservation with any id, fx 1
+SELECT * from reservation
+WHERE id = 3;
+
+--Update a reservation with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE reservation
+SET number_of_guests = 8,  meal_id = 1
+WHERE id = 4;
+
+--Delete a reservation with any id, fx 1
+DELETE FROM reservation
+WHERE id = 1;
+
+--Review
+--Get all review
+SELECT * FROM review;
+
+--Add a new review
+INSERT INTO review (title, description, meal_id, stars, created_date)
+VALUES('Perfect!', 'it was so perfect and delicious', 2, 5, '2022-08-25'),
+( 'Good!', 'tastes very nice', 3, 4,'2022-05-25'),
+("Ok!", 'Average vagetables quality', 1, 3, '2022-06-27'),
+('Amazing', 'taste was so good', 1, 5, '2022-07-18');
+
+--Get a review with any id, fx 1
+SELECT * FROM review
+WHERE id = 3;
+
+--Update a review with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE review
+SET title = 'Love it!'
+WHERE id = 4;
+
+--Delete a review with any id, fx 1
+DELETE FROM review
+WHERE id = 1;
+
+--Additional queries
+--Get meals that has a price smaller than a specific price fx 90
+Select * FROM meal
+WHERE meal.price < 260;
+
+--Get meals that partially match a title. Rød grød med will match the meal with the title Rød grød med fløde
+SELECT * FROM meal
+WHERE title LIKE 'Rød grød med%';
+
+--Get meals that has been created between two dates
+SELECT * FROM meal
+WHERE created_date BETWEEN '2022-05-22' AND '2022-07-22';
+
+--Get only specific number of meals fx return only 5 meals
+SELECT * FROM meal
+LIMIT 2;
+SELECT * FROM review;
+--Get the meals that have good reviews
+SELECT meal.title, review.stars AS starts
+FROM meal
+JOIN review
+ON review.meal_id = meal.id
+WHERE review.stars > 4;
+
+--Get reservations for a specific meal sorted by created_date
+SELECT *
+FROM reservation
+WHERE meal_id = 1
+ORDER BY created_date DESC;
+
+--Sort all meals by average number of stars in the reviews
+SELECT meal.title, SUM(review.stars) / COUNT(review.id) AS average FROM meal
+JOIN review ON review.meal_id = meal.id
+GROUP BY meal.title
+ORDER BY average;
```

Comment:
Cool! I did this with AVG function
SELECT
  meal.title,
  AVG(review.stars)
FROM
  meal
  JOIN review ON review.meal_id = meal.id
GROUP BY
  meal.title
ORDER by
  AVG(review.stars) ASC

Reply:
yes we can do like this too even its better 👍🏻